src/inmanta/env.py

```diff
@@ -61,6 +62,17 @@ class PipInstallError(Exception):
     pass
 
 
+def safe_parse_requirement(requirement_name: str) -> Requirement:
+    # Packaging Requirement is not able to parse requirements with comment. Therefore, we need to remove the `comment` part
+    drop_comment = requirement_name.split("#")[0].strip()
+    assert len(drop_comment) > 0, "The name of the requirement cannot be an empty string!"
```

Comment:
I expect this to go wrong in many places, I would also make a method to parse an entire file, that can contain empty lines and ` # `